serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt

```diff
@@ -2,82 +2,41 @@ package xyz.room409.serif.serif_shared
 import kotlinx.coroutines.*
 import io.ktor.client.*
 import io.ktor.client.request.*
+import io.ktor.client.statement.*
 import io.ktor.client.features.json.*
 import io.ktor.client.features.json.serializer.KotlinxSerializer
 import io.ktor.http.*
-import kotlinx.serialization.*
-import kotlinx.serialization.json.*
 
 sealed class Outcome<out T : Any> { }
 data class Success<out T : Any>(val value: T) : Outcome<T>()
 data class Error(val message: String, val cause: Exception? = null) : Outcome<Nothing>()
 
-@Serializable
-data class LoginRequest(val type: String, val identifier: LoginIdentifier, val password: String, val initial_device_display_name: String) {
-    constructor(username: String, password: String): this(type="m.login.password",
-                                                          identifier=LoginIdentifier(type="m.id.user",user=username),
-                                                          password=password,
-                                                          initial_device_display_name="Serif")
-}
-@Serializable
-data class LoginIdentifier(val type: String, val user: String)
-
-@Serializable
-data class LoginResponse(val access_token: String)
+// Returns the first Event of class T, or null
+inline fun <reified T> List<Event>.firstOfType(): T? = this.map { (it as? T?) }.firstOrNull { it != null }
```

Comment:
Didn't want to write out the map  lambda twice below?

Reply:
That's what I had before! Lines got reallll long